synthesize/src/main.rs

```diff
@@ -0,0 +1,46 @@
+use std::io::{self, Write};
+use std::path::PathBuf;
+
+use synthesize::Result;
+
+fn main() -> Result<()> {
+    if let Err(e) = self::cli() {
+        writeln!(io::stderr(), "{}", e)?;
+
+        std::process::exit(1);
+    }
+
+    Ok(())
```

Comment:
cc: @cole-h

Reply:
Because it makes the errors look nicer. I've attached the output below for reference.

Using your suggestion:

```
: 08:41:52 ~/w/de/bootspec/synthesize import ; cargo r /nix/var/nix/profiles/system-1-link/ out
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
     Running `/home/vin/workspace/detsys/bootspec/target/debug/synthesize /nix/var/nix/profiles/system-1-link/ out`
Error: "Failed to synthesize bootspec for /nix/var/nix/profiles/system-1-link/:\nFile exists (os error 17)"
```

Using the existing code:

```
: 08:41:59 ~/w/de/bootspec/synthesize import ; cargo r /nix/var/nix/profiles/system-1-link/ out
   Compiling synthesize v0.1.0 (/home/vin/workspace/detsys/bootspec/synthesize)
    Finished dev [unoptimized + debuginfo] target(s) in 0.35s
     Running `/home/vin/workspace/detsys/bootspec/target/debug/synthesize /nix/var/nix/profiles/system-1-link/ out`
Failed to synthesize bootspec for /nix/var/nix/profiles/system-1-link/:
File exists (os error 17)
```

Reply:
I double checked, this compiles.